### PR TITLE
Reset zoom to fit when starting drag or inserting element from library

### LINKED ISSUE #6182: New elements can appear out of view when zoomed
## Bug Description

Adding a new element by clicking on the element in the libraries will add it in a fixed position according to the general rules. When zoomed, this same thing still happens, regardless of whether this fixed position is in view.

Furthermore, the added element will always appear at default size, regardless of current zoom. This deviates from the behavior of dragging an element on to the canvas, where the resulting size will vary with the canvas.

## Expected Behavior

~The expected behavior is unknown. Maybe the current behavior is expected even?~

Edit: When the user starts dragging an element from the library or adds a new element, the Zoom is reset to Fit right away first.

## Steps to Reproduce
1. Change zoom to 200%
2. Scroll to the bottom of the page
3. Drag in an image to the canvas
4. Click the same image and observe the difference

## Screenshots
See the page preview in the bottom carousel to better understand what's happening here when the second image is added.
zoom-add-element-bug

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Context

This resets the zoom to fit regardless of what it was when inserting an element from the library by clicking or dragging. This works for all elements including:

* Local media
* Third-party media
* Text presets and text sets
* Shapes

## User-facing changes

| Before | After |
|-|-|
| click-local-before | click-local-after |
| drag-local-before | drag-local-after |
| click-text-before | click-text-after |
| click-shape-before | click-shape-after |

### QA

This PR can be tested by following these steps:

1. Set the canvas zoom to anything but fit
1. Insert any element from library by clicking or dragging
1. Observe the zoom reset to fit before the element is inserted / before dragging starts

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6182

### CODE REVIEW

File: assets/src/edit-story/components/canvas/useInsertElement.js [hunk 1]
```diff
@@ -85,6 +89,7 @@ function useInsertElement() {
    */
   const insertElement = useCallback(
     (type, props) => {
+      setZoomSetting(ZOOM_SETTING.FIT);
```
Comment (barklund) [inline assets/src/edit-story/components/canvas/useInsertElement.js:94]: Note that this is only required for the text presets, as they cannot be dragged in. Any element that can be dragged in will actually reset the canvas zoom in the `dragStart` handler on the draggable element also for a click.

File: assets/src/edit-story/components/library/panes/shared/libraryMoveable.js [hunk 2]
```diff
@@ -76,10 +77,15 @@ function LibraryMoveable({
   const overlayRef = useRef(null);
   const moveable = useRef(null);
 
+  // These useLayout's are specifically kept as separate entries
```
Comment (miina) [inline assets/src/edit-story/components/library/panes/shared/libraryMoveable.js:80]: Hm, what would happen when using the same `useLayout`?

**Review by miina — APPROVED** [review]
🥇  So much more clear now when adding the elements.

**Comment by miina** [comment]
@barklund The tests and lints are failing now.

File: assets/src/edit-story/components/library/panes/shared/libraryMoveable.js (hunk 2 above)
Comment (barklund) [reply assets/src/edit-story/components/library/panes/shared/libraryMoveable.js:80]: Nothing out of the ordinary, it's just the same function invoked twice with different arguments.

**Pull request merged**